### PR TITLE
Integrate repo with snyk GHA

### PR DESCRIPTION
**Pull request by NovemberTang**
This PR reliably integrates the repository with the snyk GitHub action. This PR has only been raised on repos that have already been tested to make sure scanning will work out of the box. ‘reliably integrated’ means that this action compares the hash of the last commit on main to the one that snyk has, and makes sure that they match.

### CODE REVIEW

**Comment by johnduffell** [comment]
> ‘reliably integrated’ means that this action compares the hash of the last commit on main to the one that snyk has, and makes sure that they match.

Oh nice, this has been a big problem for a long time and snyk have basically shrugged their shoulder when I raised it in the past!  So I'm really glad it's being solved by you 👏

**Review by johnduffell — APPROVED** [review]
hard to test an action that only runs on push to main, but let's merge it and see what happens!

File: .github/workflows/snyk.yml
Comment (paulbrown1982) [inline .github/workflows/snyk.yml:7]: We will need to migrate this repo over before we can merge this, or we can merge and do that afterwards. What do you think @NovemberTang ?

Comment (kenoir) [reply .github/workflows/snyk.yml:7]: I'm helping @NovemberTang with the Snyk migrations. I think we should hold off until the default branch has been updated. It looks like this project will need to updated in TeamCity to do so.

Comment (kenoir) [reply .github/workflows/snyk.yml:7]: I've changed my mind ... no need to couple these things.

**Pull request merged**
